paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -715,18 +715,55 @@ internal class PaymentSheetViewModelTest {
         assertThat(isEnabled).isTrue()
 
         viewModel.savedStateHandle[SAVE_PROCESSING] = true
+
+        idleLooper()
+
         assertThat(isEnabled).isFalse()
 
         viewModel.savedStateHandle[SAVE_PROCESSING] = false
+
+        idleLooper()
+
         assertThat(isEnabled).isTrue()
 
         viewModel.setEditing(true)
+
+        idleLooper()
+
         assertThat(isEnabled).isFalse()
 
         viewModel.setEditing(false)
+
+        idleLooper()
+
         assertThat(isEnabled).isTrue()
     }
 
+    @Test
+    fun `buttonsEnabled should be true when not processing and not editing`() = runTest {
+        val viewModel = createViewModel()
+
+        viewModel.buttonsEnabled.test {
+            assertThat(awaitItem()).isTrue()
+
+            viewModel.setEditing(true)
+            idleLooper()
+
+            assertThat(expectMostRecentItem()).isFalse()
```

Comment:
how come we use `expectMostRecentItem`. My guess is because we're emitting a bunch of intermediate events. Can we make the emissions be just the ones we want?

Reply:
Since the flow is combined, it will update whenever any of the inner flows are updated, in this case `editing` and `processing` both change, so I am using `expectMostRecentItem` here. Is there a better way to do this?

Reply:
Ideally we would just have an assert that it changed the same number of items it was updated. `awaitItem` calls for each change (even if it means it's value didn't change, or is an intermediate value).